core/new-gui/src/app/workspace/component/workspace.component.ts

```diff
@@ -135,33 +121,27 @@ export class WorkspaceComponent implements OnInit, OnDestroy {
         this.workflowPersistService.persistWorkflow(this.workflowActionService.getWorkflow())
           .subscribe((updatedWorkflow: Workflow) => {
             this.workflowActionService.setWorkflowMetadata(updatedWorkflow);
-            this.workflowCacheService.setCacheWorkflow(this.workflowActionService.getWorkflow());
             this.location.go(`/workflow/${updatedWorkflow.wid}`);
           });
         // to sync up with the updated information, such as workflow.wid
       }
     }));
   }
 
-  private loadWorkflowWithID(id: number): void {
-    this.subscriptions.add(this.workflowPersistService.retrieveWorkflow(id).subscribe(
+  private loadWorkflowWithId(wid: number): void {
```

Comment:
I suggest we move `disableWorkflowModification` here, instead of doing it at the caller. In this way, whoever calls this function doesn't need to worry about disable workflow modification.

Reply:
Also can we add a "loading" sign somewhere so user knows? no need to do it in this PR, we can do it later.

Reply:
we can definitely add a loading badge! will do in future PR.